run_iwyu_tests.py

```diff
@@ -134,8 +140,16 @@ def PrintLoadedTestsAndFiles():
       # https://mesonbuild.com/Unit-tests.html#skipped-tests-and-hard-errors
       print('Skipped %s: %s' % (runner_args.run_test_file, e))
       exit(77)
+    except AssertionError as e:
+      if xfail:
+        print("%s: Expected failure" % runner_args.run_test_file)
+        exit(0)
+      raise e from None
```

Comment:
You can just say `raise` to re-throw the active exception, no need for `from None` (though thanks for demonstrating that, I hadn't seen the construct before!)